### PR TITLE
DOPS-216: Add MM_INSTALL_TYPE=kubernetes

### PR DESCRIPTION
**Pull request by metanerd**
#### Summary
 Sets an environment variable to get an idea of the preferred installation type. 

#### Ticket Link

https://mattermost.atlassian.net/jira/software/projects/DOPS/boards/66/backlog?selectedIssue=DOPS-216

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: pkg/mattermost/mattermost.go [hunk 1]
```diff
@@ -371,6 +371,10 @@ func GenerateDeployment(mattermost *mattermostv1alpha1.ClusterInstallation, dbIn
 			Name:  "MM_CLUSTERSETTINGS_CLUSTERNAME",
 			Value: "production",
 		},
+		{
+			Name:  "MM_INSTALL_TYPE",
+			Value: "operator",
```
Comment (metanerd) [inline pkg/mattermost/mattermost.go:376]:
```suggestion
			Value: "kubernetes",
```

**Review by gabrieljackson — APPROVED** [review]
Looks good!

I did want to check that we want to use `kubernetes` as the value for this though. We have at least two types of kubernetes deployments with helm and the operator. Do we want to be more specific with something like `kubernetes-operator` or something similar?

**Comment by metanerd** [comment]
@grundleborg ^

**Comment by grundleborg** [comment]
@gabrieljackson @metanerd making it more specific in that case would be great - you can choose the term (so `kubernetes_operator` is great).

**Review by cpanato — CHANGES_REQUESTED** [review]
@metanerd can you update the variable name per George request?
otherwise lgtm

File: pkg/mattermost/mattermost.go [hunk 2]
```diff
@@ -371,6 +371,10 @@ func GenerateDeployment(mattermost *mattermostv1alpha1.ClusterInstallation, dbIn
 			Name:  "MM_CLUSTERSETTINGS_CLUSTERNAME",
 			Value: "production",
 		},
+		{
+			Name:  "MM_INSTALL_TYPE",
+			Value: "kubernetes",
```
Comment (metanerd) [inline pkg/mattermost/mattermost.go:376]:
```suggestion
			Value: "kubernetes-operator",
```

**Review by cpanato — APPROVED** [review]

**Comment by cpanato** [comment]
adding @grundleborg as a reviewer as well since he asks for the change

**Review by grundleborg — APPROVED** [review]
LGTM

**Pull request merged**